lib/Target/CBackend/CBackend.cpp

```diff
@@ -493,6 +493,18 @@ raw_ostream &CWriter::printSimpleType(raw_ostream &Out, Type *Ty,
   }
 }
 
+raw_ostream &
+CWriter::printTypeNameForAddressableValue(raw_ostream &Out, Type *Ty, bool isSigned) {
+  if (!isEmptyType(Ty))
+    return printTypeName(Out, Ty, isSigned);
+  else {
+    // We can't directly declare a zero-sized variable in C, so we have to
+    // use a single-byte type instead, in case a pointer to it is taken.
+    // We can then fix the pointer type in writeOperand.
+    return Out << "char /* (empty) */";
+  }
```

Comment:
style nit: I believe LLVM style omits all unnecessary `{}`?

Reply:
Indeed, but my intuition is this doesn't apply if there's a comment, and [the coding standard seems to agree](https://llvm.org/docs/CodingStandards.html):

> We consider that readability is harmed when omitting the brace in the presence of a single statement that is accompanied by a comment

However, that standard also suggests hoisting comments to avoid this problem, so I will do that.